packages/mdc-select/foundation.ts

```diff
@@ -87,6 +89,12 @@ export class MDCSelectFoundation extends MDCFoundation<MDCSelectAdapter> {
     this.helperText_ = foundationMap.helperText;
   }
 
+  /** Returns the index of the currently selected menu item, or -1 if none. */
+  getSelectedIndex(): number {
+    const selectedMenuItem = this.adapter_.getSelectedMenuItem();
```

Comment:
The state of selectedIndex needs to be stored in foundation. Currently seems like we're using DOM as source of truth to retrieve selected index.

See MDCListFoundation for example.

Reply:
Will apply changes to PR #4823, since its changes were partially merged into this one (on accident). Sorry for the confusion!